### PR TITLE
Implement error policy

### PR DESCRIPTION
**Pull request by mainawycliffe**
Attempt to implement errorPolicy, to address this [issue](https://github.com/zino-app/graphql-flutter/issues/143)

### Breaking changes

- No breaking changes, should work expected without code being updated!

### CODE REVIEW

File: packages/graphql/lib/src/core/query_manager.dart [hunk 1]
```diff
@@ -269,23 +271,61 @@ class QueryManager {
...
+
+  bool _shouldReturnData(FetchResult fetchResult, ErrorPolicy errorPolicy) {
+    // if there are no errors, return data
+    if (fetchResult.errors == null) {
+      return true;
+    }
+
+    // if there is data and errors, return data if the error policy allows it
+    // if errorPolicy is return all or ignore errors, return data
+    if (fetchResult.errors != null &&
```
Comment (micimize) [inline packages/graphql/lib/src/core/query_manager.dart]: for this and in `_shouldReturnErrors` I think a switch statement would be clearer:
```dart
switch (options.errorPolicy) {
  case ErrorPolicy.ignore:
  case ErrorPolicy.all:
    return false;
  case ErrorPolicy.none:
  default:
    return true;
}
```
but that's a pretty minor thing - overall this is great and I'm green lighting it 👍

**Comment by micimize** [comment]
@mainawycliffe leaving open for a bit if you want to make the `switch`, otherwise just comment and I'll merge

**Comment by mainawycliffe** [comment]
@micimize yes, i agree switch case is much clearer. I will make the necessary changes.

**Pull request merged**